File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    total = 0

    for i in range(number + 1):
        total += i

    return pow(total, 2)

def sum_of_squares(number):
    total = 0

    for i in range(number + 1):
        total += pow(i, 2)

    return total

def difference_of_squares(number):
    square_of_the_sum = square_of_sum(number)
    sum_of_the_squares = sum_of_squares(number)

    return square_of_the_sum - sum_of_the_squares

File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
export class Squares {
  constructor(number) {
    this._number = number;
  }

  get sumOfSquares() {
    let total = 0;
    for (let i = 1; i <= this._number; i++) {
      total += Math.pow(i, 2);
    }
    return total;
  }

  get squareOfSum() {
    let sum = 0;
    for (let i = 1; i <= this._number; i++) {
      sum += i;
    }
    return sum * sum;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
